Python/bltinmodule.c

```diff
@@ -1449,6 +1481,43 @@ map_next(mapobject *lz)
         PyMem_Free(stack);
     }
     return result;
+check:
+    if (PyErr_Occurred()) {
+        if (!PyErr_ExceptionMatches(PyExc_StopIteration)) {
+            // next() on argument i raised an exception (not StopIteration)
+            return NULL;
+        }
+        PyErr_Clear();
+    }
+    if (i) {
+        // ValueError: map() argument 2 is shorter than argument 1
+        // ValueError: map() argument 3 is shorter than arguments 1-2
```

Comment:
At a first glance, 1-2 looks like “one minus two”.
Would “previous argument[s]” would work better?

Reply:
This matches the exception of `zip()`.